### ISSUE TITLE
Making a private dashboard with private insights more visible should handle the privacy explicitly. 

### ISSUE THREAD

**Issue by Joelkw** [sourcegraph/sourcegraph#23003]
Right now it seems that I can make a private dashboard, add private insights, then make it org-visible and it carries the insights with it. Yet the insight itself appears to still be private (if I dig into the settings of it via clicking "edit" insight). 

I think if you try to make a dashboard with a private insight more visible, we should either a) make the private insight org visible AND add an alert/confirm letting you know this is going to happen, or b) let you know in an alert/confirm if you make this dashboard public it will strip the private insights. I prefer a) because it enables smoother sharing, and presumably a user doesn't want to make a dashboard org visible in the first place if they think it's too sensitive. 

Here's a loom showing the behavior as it is now: 
https://www.loom.com/share/76fc2bb0089a43ebbc627b9a5700f63d

**Comment by Joelkw** [comment]
(I do **not** expect this makes the 3.30 release, of course. No worries on that.) 

@AlicjaSuska @vovakulikov @felixfbecker thoughts here? I think we need a design for either a) or b) options

**Comment by Joelkw** [comment]
I verified that the insight remains private, so different users see different sets of insights on the dashboard. I think the expectation is that dashboards are the same insight set for all users, but I can dig more into that with customers after we launch dashboards!

**Comment by AlicjaSuska** [comment]
I agree that we should go with the option a. I've put together a [design](https://www.figma.com/file/5ahYGyPdTGS9NVdMdRXe4m/Making-a-private-dashboard-with-private-insights...-23003?node-id=246%3A11) - we can add information under the selected option in 'Configure dashboard' screen. I think it's enough to communicate it and the information is provided in the context.

@Joelkw @vovakulikov

**Comment by AlicjaSuska** [comment]
To communicate better the visibility changes we should:
- Case 1: when extending the visibility of the dashboard (private to org/global and org to global): display a note about expanding the visibility of all insights included in the dashboard.
- Case 2: when limiting the visibility of the dashboard (org/global to private and global to org): display checkbox to give users choice between changing the visibility settings for all the insights included in the dashboard ...or leaving insights' visibility settings as they are and change only dashboard settings.

Scenarios and design spec:

1. [Convert PRIVATE to ORG or GLOBAL](https://www.figma.com/file/5ahYGyPdTGS9NVdMdRXe4m/Making-a-private-dashboard-with-private-insights...-23003?node-id=949%3A4850) (extending visibility)
- cannot keep insight private, we have to change permissions to Org/Global.
- display **note**: 'The visibility setting of all the insights on this dashboard will change to [org/global].'

2. [Convert ORG or GLOBAL to PRIVATE](https://www.figma.com/file/5ahYGyPdTGS9NVdMdRXe4m/Making-a-private-dashboard-with-private-insights...-23003?node-id=275%3A1628) (limiting visibility)
- ee can give user a choice between:
a. keeping insights' current visibility settings and only make the dashboard private
b. or... converting all the insights to private (and potentially deleting them from other dashboards)
- display **checkbox**: 'Change the visibility setting of all the insights on this dashboard to private. This may remove insights from other dashboards.'

3. [Convert GLOBAL to ORG](https://www.figma.com/file/5ahYGyPdTGS9NVdMdRXe4m/Making-a-private-dashboard-with-private-insights...-23003?node-id=949%3A3722) (limiting visibility)
- we can give user a choice between:
a. keeping insights' current visibility settings and change the dashboard settings to Org
b. or... converting all the insights to org (and potentially deleting them from other dashboards)
- display **checkbox**: 'Change the visibility setting of all the insights on this dashboard to [org name]. This may remove insights from other dashboards.'

@Joelkw I've discussed this solution with @felixfbecker before his vacation. Could you please review this final approach? If everything is fine, I'll move to 'ready for dev'.

**Comment by vovakulikov** [comment]
@AlicjaSuska Your cases sound good to me. I would add another case 

**Move Dashboard from one org to another org**. 
In this case, we also can give 2 options to users
- Keeping insights from this dashboard with the same visibility setting and then some insight probably will stay in the same org and they won't be available in this dashboard then we move this to another org 
- Moving insight with the dashboard to another org. In this case, all insights in this org dashboard will be moved in receiver-org with the moving dashboard.

**Comment by AlicjaSuska** [comment]
Good point @vovakulikov !

I think that in the org to org case we should transfer the full dashboard (with all the insights) to another org. We should avoid the situation of not transferring all the insights in the dashboard.

So in this case (transfer dashboard from ORG A to ORG B), I'd suggest giving a choice between:
a. adding ORG B to all the insights on the dashboard, keep all the visibility settings of each insight as they were
b. changing visibility settings of all the insights to ORG B only

This would cover 2 scenarios:
a. -> I want to transfer the dashboard but don't want to affect insights visibility, it's not a privacy concern, I'm just reorganizing dashboards
b. -> I'm concerned about privacy, I want this dashboard and its insights to be visible ONLY to Org B

wdyt about it @vovakulikov ?

**Comment by vovakulikov** [comment]
@AlicjaSuska so from org to org we also have to have some checkbox with precaution messages about that you can move insights together with the dashboard or you can move only the dashboard and leave insights as they are, right?

**Comment by Joelkw** [comment]
@vovakulikov @AlicjaSuska things have changed a bit (now you can add a private insight to an org dashboard, right?) but we might still want to notify the user they're about to do that?

**Comment by vovakulikov** [comment]
Based on the latest sync with @AlicjaSuska for now we should focus more on https://github.com/sourcegraph/sourcegraph/issues/29943 and leave the broader discussions about sharing/permission for later.
